src/doc/rustc/src/platform-support.md

```diff
@@ -148,13 +150,13 @@ target | std | notes
 `armv7r-none-eabi` | * | Bare ARMv7-R
 `armv7r-none-eabihf` | * | Bare ARMv7-R, hardfloat
 `asmjs-unknown-emscripten` | ✓ | asm.js via Emscripten
-`i586-pc-windows-msvc` | * | 32-bit Windows w/o SSE
-`i586-unknown-linux-gnu` | ✓ | 32-bit Linux w/o SSE (kernel 3.2, glibc 2.17)
-`i586-unknown-linux-musl` | ✓ | 32-bit Linux w/o SSE, MUSL
-[`i686-linux-android`](platform-support/android.md) | ✓ | 32-bit x86 Android
-`i686-unknown-freebsd` | ✓ | 32-bit FreeBSD
-`i686-unknown-linux-musl` | ✓ | 32-bit Linux with MUSL
-[`i686-unknown-uefi`](platform-support/unknown-uefi.md) | * | 32-bit UEFI
+`i586-pc-windows-msvc` | * | 32-bit Windows w/o SSE [^x86_32-floats-noSSE]
+`i586-unknown-linux-gnu` | ✓ | 32-bit Linux w/o SSE (kernel 3.2, glibc 2.17) [^x86_32-floats-noSSE]
+`i586-unknown-linux-musl` | ✓ | 32-bit Linux w/o SSE, MUSL [^x86_32-floats-noSSE]
+[`i686-linux-android`](platform-support/android.md) | ✓ | 32-bit x86 Android [^x86_32-floats-SSE]
+`i686-unknown-freebsd` | ✓ | 32-bit FreeBSD [^x86_32-floats-SSE]
+`i686-unknown-linux-musl` | ✓ | 32-bit Linux with MUSL [^x86_32-floats-SSE]
+[`i686-unknown-uefi`](platform-support/unknown-uefi.md) | * | 32-bit UEFI [^x86_32-floats-SSE]
```

Comment:
The uefi targets use softfloat by default

https://doc.rust-lang.org/nightly/rustc/platform-support/unknown-uefi.html

Reply:
That page doesn't talk about flaots.

So what does this mean? The x87 registers and instructions are completely unused and therefore we don't have FP issues on these targets?

Reply:
Ah, I searched for "float" and didn't find that -- thanks.

Though I don't know what "FP unit is disabled" means. What happens when a function returns an `f32`, how is the return value passed to the caller?

Reply:
It is passed using the soft float ABI, which varies slightly from target to target but typically means that it is given the struct or integer calling convention and thus typically means it is either potentially in-register or always dropped on the stack.

Reply:
Okay, so no x87 shenanigans should be possible then -- thanks.